### ISSUE TITLE
Do not try to propose a fix when there is an uppercase in the middle of a word

### ISSUE THREAD

**Issue by sylvestre** [lucasdemarchi/codespell#314]
False positive as example: _aParent => apparent_
When there is an uppercase in the middle of the word, maybe codespell should not try to correct it.

**Comment by luzpaz** [comment]
Yes, good idea. I think trying to flesh out the conditions for this is important.

**Comment by sylvestre** [comment]
What about ignoring like this?
`m = re.match(r'^[a-z][A-Z][a-z]*', typo)`

**Comment by jankeromnes** [comment]
> `m = re.match(r'^[a-z][A-Z][a-z]*', typo)`

I think this wouldn't catch variables like `anInteger`, `itemCounter`, `annualGrowthRate`, etc.

**Comment by peternewman** [comment]
What's the use case here? Personally if camel casing, I'd like to spell-check item and Counter and ensure they're valid words. Although if you're doing the style where you prepend the type of the variable (int, bool etc) e.g. iCounter, bProcessed, that probably falls down with some types as b won't be a valid word on it's own.

**Comment by mikedh** [comment]
It also attempts to add a space in some cases which is somewhat undesirable and breaking in variable/class names:
```
class FileTests(g.unittest.TestCase):
	FileTests  ==> file tests (Y/n) 
```
